Compilation fails on Debian testing

Using `make cqueues5.3` on a Debian Stretch system with OpenSSL 1.0.2 fails with the following error:

```
~/cqueues/src/lib/socket.c: In function ‘so_starttls’:
~/cqueues/src/lib/socket.c:2089:53: warning: implicit declaration of function ‘SSLv3_server_method’ [-Wimplicit-function-declaration]
  if (!method->ssl_connect || method->ssl_connect == SSLv3_server_method()->ssl_
                                                     ^
~/cqueues/src/lib/socket.c:2089:74: error: invalid type argument of ‘->’ (have ‘int’)
 !method->ssl_connect || method->ssl_connect == SSLv3_server_method()->ssl_conne
                                                                     ^
~/cqueues/src/lib/GNUmakefile:37: recipe for target '~/cqueues/src/lib/socket.o' failed
make: *** [~/cqueues/src/lib/socket.o] Error 1
```

I'm guessing because libssl was built with the `no-ssl3` option.

--- a/src/lib/socket.c
+++ b/src/lib/socket.c
@@ -2078,15 +2078,15 @@ int so_starttls(struct socket *so, const struct so_starttls *cfg) {
 	SSL_set_mode(so->ssl.ctx, SSL_MODE_ENABLE_PARTIAL_WRITE);
 
 	/*
-	 * NOTE: SSLv3_server_method()->ssl_connect should be a reference to
+	 * NOTE: SSLv23_server_method()->ssl_connect should be a reference to
 	 * OpenSSL's internal ssl_undefined_function().
 	 *
-	 * Server methods such as SSLv23_server_method(), etc. should have
+	 * Server methods such as TLSv1_2_server_method(), etc. should have
 	 * their .ssl_connect method set to this value.
 	 */
 	method = SSL_get_ssl_method(so->ssl.ctx);
 
-	if (!method->ssl_connect || method->ssl_connect == SSLv3_server_method()->ssl_connect)
+	if (!method->ssl_connect || method->ssl_connect == SSLv23_server_method()->ssl_connect)
 		so->ssl.accept = 1;
 
 	if (!so->ssl.accept && so->opts.tls_sendname && so->opts.tls_sendname != SO_OPTS_TLS_HOSTNAME) {

